ui/src/message_popup/linear_session/linear_session.jsx

```diff
@@ -57,9 +69,13 @@ export default React.createClass({
   renderContent() {
     const {questions} = this.props;
     const {responses} = this.state;
-    if (responses.length >= questions.length) return this.props.summaryEl(questions, responses);
 
-    const question = questions[responses.length];
+    if (this.props.getNextQuestion === undefined || this.props.getNextQuestion(questions, responses) === null) {
```

Comment:
Ah got it, thanks for explaining!  Yeah seems like an edge case bug between React and Flow, where Flow can't tell React will guarantee that `this.props.getNextQuestion` will have a value by that point.

For these "possibly undefined values" warnings you can explicitly check that it's defined with `const question = this.props.getNextQuestion && this.props.getNextQuestion(...)`.

For those other warnings, looks like just a `handleLayout: ` with a colon instead of an equals should help.  If those don't work then feel free to drop another comment or we can take a look together.